### PR TITLE
Add analytics service to v3

### PR DESCRIPTION
**Pull request by crookedneighbor**
Ports v2 analytics service as middleware for v3.

### CODE REVIEW

**Comment by paglias** [comment]
There are a few small things I'm not convinced about, I'll comment once I get to a computer later but wait before merging

**Comment by crookedneighbor** [comment]
Sure thing.

File: website/src/middlewares/api-v3/analytics.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    };
+  } else {
+    service = mockAnalyticsService;
+  }
+
+  return service;
+}
+
+export default function attachAnalytics (req, res, next) {
+  res.analytics = _generateService();
```
Comment (paglias) [inline website/src/middlewares/api-v3/analytics.js]: The _generateService() function should only run once, not on every request since it's always the same

Comment (crookedneighbor) [inline website/src/middlewares/api-v3/analytics.js]: I was thinking that too. This was mainly for being able to test prod and non-prod. We could memo-ize it.

Comment (crookedneighbor) [inline website/src/middlewares/api-v3/analytics.js]: Or we could just not care about testing prod vs non-prod for this.

Comment (paglias) [inline website/src/middlewares/api-v3/analytics.js]: I would say in this case better to prioritize speed

**Comment by crookedneighbor** [comment]
Merged with https://github.com/HabitRPG/habitrpg/commit/cdb05e1b424f74be10c67a448e151a4944b6882d

**Pull request closed without merging**